### ISSUE TITLE
Can't see hints after 0.3.8

### ISSUE THREAD

**Issue by john-kurkowski** [Morantron/tmux-fingers#32]
`git bisect` lead me to the first bad commit, 844d6b831515d9b8d846889371f18f66529b52ec.

| Before | After applying 844d6b8 |
| --- | --- |
| ![screen shot 2017-05-18 at 11 57 10 am](https://cloud.githubusercontent.com/assets/299487/26218910/939cc880-3bc1-11e7-985b-a64c5db7eec3.png) | ![screen shot 2017-05-18 at 11 56 35 am](https://cloud.githubusercontent.com/assets/299487/26218935/b05b032e-3bc1-11e7-981f-8f5bea3d9eee.png) |

**Comment by Morantron** [comment]
Huh, weird! :thinking: 

Have you tried to source again your `.tmux.conf` after upgrading?

What does the following return?

```
gawk -W version
tmux -V
echo $SHELL
echo $TERM
```

**Comment by john-kurkowski** [comment]
```zsh
GNU Awk 4.1.4, API: 1.1 (GNU MPFR 3.1.5, GNU MP 6.1.1)
Copyright (C) 1989, 1991-2016 Free Software Foundation.

This program is free software; you can redistribute it and/or modify
it under the terms of the GNU General Public License as published by
the Free Software Foundation; either version 3 of the License, or
(at your option) any later version.

This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU General Public License for more details.

You should have received a copy of the GNU General Public License
along with this program. If not, see http://www.gnu.org/licenses/.
tmux 2.3
/usr/local/bin/zsh
screen-256color
```

**Comment by john-kurkowski** [comment]
Yes, I re-sourced .tmux.conf in between each step of the bisect.

**Comment by Morantron** [comment]
Ooooh shit, I think I now what the issue might be.

Can you please try this?

```sh
cd /path/to/tmux-fingers
vendor/tmux-printer/tmux-printer "#[fg=red,bold]boom!"
```
tmux-printer is a git submodule, make sure it's updated properly. Looks like https://github.com/morantron/tmux-fingers is pointing to `awk`, but it should be using `gawk` instead.

If that's the issue version `0.5.0` should work for you

**Comment by john-kurkowski** [comment]
I don't think that `!` is safe in double quotes..? Switching to single quotes gets you a red "boom!"

![screen shot 2017-05-20 at 10 49 07 am](https://cloud.githubusercontent.com/assets/299487/26278232/37bb9160-3d4a-11e7-8792-267e32edf41a.png)

> tmux-printer is a git submodule, make sure it's updated properly

How do I do that? I update tmux-fingers via tpm's `Prefix + U`. Maybe tpm doesn't support submodules? (Can you use a subtree instead?)

**Comment by john-kurkowski** [comment]
> tmux-printer is a git submodule, make sure it's updated properly ... version `0.5.0` should work for you

Through some combination of [this SO answer on updating submodules](http://stackoverflow.com/a/21195182/62269) and cleaning untracked files, 0.5.0 does indeed show hints again! Just like 0.3.8. However, yanking to my clipboard stops working.

**Comment by john-kurkowski** [comment]
Any reason to stop at 0.5.0? FYI, if I keep going to 0.6.1, the hint color bleeds into the targets. And yanking to my clipboard still has no effect.

![screen shot 2017-05-20 at 11 04 34 am](https://cloud.githubusercontent.com/assets/299487/26278314/58058848-3d4c-11e7-9d7a-dfe5bcd05fbe.png)

**Comment by Morantron** [comment]
> How do I do that? I update tmux-fingers via tpm's Prefix + U. Maybe tpm doesn't support submodules? (Can you use a subtree instead?)

AFAIK tpm supports submodules, I'll make some tests to ensure upgrading from a prior version works though.

> if I keep going to 0.6.1, the hint color bleeds into the targets

Weee, at least hints are shown now! :B Did you run the commands [here](https://github.com/Morantron/tmux-fingers/issues/32#issuecomment-302867920) within a tmux session?

That's usually because of a TERM value that does not support all fancy tput escape sequences. In theory `set -g default-terminal screen-256color` in your `.tmux.conf` should fix it.

If not, you can customize the appearance of the hints. Check docs [here](https://github.com/morantron/tmux-fingers#fingers-hint-format).

Anyway, looks like more sensible defaults should be provided so the visibility is better regardless of the TERM set.

> However, yanking to my clipboard stops working

In theory that was broken in `0.5.0` but fixed in `0.6.1`.

Summing up, `0.6.1` with proper TERM setting or custom colors should work for you! Both readable hints and clipboard integration.

If that fixes the issue for you I'll close this issue and create two new ones to address:

* Improve hint format defaults
* Check that upgrade to 0.6.0 works. Add submodule check in health check script.

Thanks!

**Comment by john-kurkowski** [comment]
> Did you run the commands

Yep, see [this comment](https://github.com/Morantron/tmux-fingers/issues/32#issuecomment-302886035).

**Comment by john-kurkowski** [comment]
> If not, you can customize the appearance of the hints.

Customizing `@fingers-hint-format` did the trick for 0.6.1, letting me see hints.

Copying to my clipboard still not working in 0.6.1. Maybe there's a 3rd issue then? Note I've been using the latest tmux-yank this whole time. Going back to 0.3.8 fixes everything.

**Comment by Morantron** [comment]
Clipboard _works on my machine_ :tm: with tmux-yank's SHA 21c3ba7f846cc7f114548c4f9de8d6dc36f06990

I need to check with latest master.

**Comment by john-kurkowski** [comment]
When I set aside the hint rendering problem, trying to pin down the clipboard problem, `git bisect` lead me to 76b187581e473c743dd26328d539c8915648e004. I think the spirit of that commit was in the right direction. And it's wildly different with the latest `master`. So I stopped investigating that commit. I turned my attention to `master`. Let's roll forward rather than backwards. That lead me to 2 macOS-specific issues: #33 & #34.

**Comment by Morantron** [comment]
Cool cool cool, created two new issues to track the other weird stuff we found out. Thank you! :raised_hands: 

https://github.com/Morantron/tmux-fingers/issues/36
https://github.com/Morantron/tmux-fingers/issues/35
